src/main/java/org/spongepowered/api/text/chat/ChatVisibility.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.chat;
+
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.text.translation.Translatable;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+/**
+ * Represents a visibility mode for a client's chat.
+ */
+@CatalogedBy(ChatVisibilities.class)
+public interface ChatVisibility extends CatalogType, Translatable {
```

Comment:
@JBYoshi Well yes, that's expected (thought `isChat` isn't a very good name, it's the opposite since chat is `0`):
```java
        if (packetIn instanceof S02PacketChat)
        {
            S02PacketChat chatPacket = (S02PacketChat)packetIn;
            EntityPlayer.EnumChatVisibility visibility = this.playerEntity.getChatVisibility();

            if (visibility == EntityPlayer.EnumChatVisibility.HIDDEN)
            {
                return;
            }

            if (visibility == EntityPlayer.EnumChatVisibility.SYSTEM && !chatPacket.isChat())
            {
                return;
            }
        }
```
```java
    public boolean isChat()
    {
        return this.type == 1 || this.type == 2;
    }
```